RELEASE.md

```diff
@@ -7,6 +7,18 @@
 ## Breaking changes to the API
 Merged the ExistsMixin into AbstractDataSet
 
+# [Upcoming] Release 0.14.2
```

Comment:
Our next release is 0.15.0, so could you please add your change in there? Also please add your name and Github link in the contributor for 0.15 release.